src/configuration/remapper.ts

```diff
@@ -76,29 +88,194 @@ export class Remapper implements IRemapper {
     vimState: VimState
   ): Promise<boolean> {
     this._isPotentialRemap = false;
+    this._allowPotentialRemapOnFirstKey = vimState.recordedState.allowPotentialRemapOnFirstKey;
+    let allowBufferingKeys = true;
+    let remainingKeys: string[] = [];
 
     if (!this._remappedModes.includes(vimState.currentMode)) {
       return false;
     }
 
     const userDefinedRemappings = configuration[this._configKey] as Map<string, IKeyRemapping>;
 
+    if (keys[keys.length - 1] === '<BufferedKeys>') {
+      // Timeout finished. Don't let an ambiguous or potential remap start another timeout again
+      keys = keys.slice(0, keys.length - 1);
+      allowBufferingKeys = false;
+    }
+
+    if (keys.length === 0) {
+      return true;
+    }
+
     this._logger.debug(
       `trying to find matching remap. keys=${keys}. mode=${
         Mode[vimState.currentMode]
       }. keybindings=${this._configKey}.`
     );
+
     let remapping: IKeyRemapping | undefined = this.findMatchingRemap(
       userDefinedRemappings,
       keys,
       vimState.currentMode
     );
 
+    let isPotentialRemap = false;
+    let isPotentialRemapOnOtherRemappings = false;
+    // Check to see if a remapping could potentially be applied when more keys are received
+    const keysAsString = keys.join('');
+    if (keysAsString !== '') {
+      for (let remap of userDefinedRemappings.keys()) {
+        if (remap.startsWith(keysAsString) && remap !== keysAsString) {
+          isPotentialRemap = true;
+          break;
+        }
+      }
+    }
+    if (!isPotentialRemap) {
+      // Check the other remappings for this mode
+      let otherConfigKey: string;
```

Comment:
I'd rather `otherConfigKey` be calculated on `Remapper` construction, and stored as a property